cdap-app-templates/cdap-etl/cdap-etl-proto/src/main/java/io/cdap/cdap/etl/proto/v2/spec/StageSpec.java

```diff
@@ -118,6 +133,7 @@ public boolean equals(Object o) {
     return Objects.equals(name, that.name) &&
       Objects.equals(plugin, that.plugin) &&
       Objects.equals(inputSchemas, that.inputSchemas) &&
+      Objects.equals(HashMultiset.create(inputStages), HashMultiset.create(that.inputStages)) &&
```

Comment:
Why need a Multiset for equals? Because we can contains duplicates in input stages?

Reply:
yes, it's handle duplicates input stage names (not sure if that can actually happen).

Reply:
changed MultiSet -> Set